### DISCUSSION TITLE
Output new/updated formulae & casks one per line

### DISCUSSION THREAD

**Discussion by rgoldberg** [Homebrew/discussions#3372] [Everyday usage] [answered]
Is it possible for commands like `brew update` to output new/updated formulae & casks one per line?

If so, how?

If not, is there an existing feature request issue? I didn't see any in the Homebrew/brew repo. Also, if not, can anyone show me where the output is generated, flags are processed, etc., so I can make a PR, if you'd accept one.

**Comment by gromgit** [comment, marked as answer]
Try piping the output to another program like `less`, or redirect it to a file. You might be surprised. :grinning:

**Comment by rgoldberg** [comment]
`brew update | more -e` didn't change the output to one formula/cask per line.

I'm running the command in a zsh 5.9 script (with shebang `#!/usr/bin/env zsh`) on macOS 12.4, using `/usr/bin/more`.

**Reply by gromgit** [reply]
That's probably because `brew update` output is going to _stderr_ rather than stdout, so `brew` is still seeing a terminal on its chosen output file descriptor:
```
box1$ brew update | tee /tmp/update.log
Updated Homebrew from b94147021 to 150c7d4d4.
Updated 2 taps (homebrew/cask and homebrew/core).
==> New Formulae
fastnetmon
==> New Casks
betterandbetter                              betterdisplay                                dmg-canvas                                   fly-key                                      ui                                           wow
==> Outdated Formulae
alsa-lib          angband           aom               binutils          direnv            gcc               imagemagick       libidn            libpq             libxml2           pylint            speexdsp          sqlite-utils      storj-uplink      vim

You have 15 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

box2$ brew update |& tee /tmp/update.log
Updated Homebrew from 4103f9dd0 to 150c7d4d4.
Updated 2 taps (homebrew/core and homebrew/cask).
==> New Casks
betterdisplay
ui
wow
==> Outdated Formulae
aom
direnv

You have 2 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.
```
